tfe/resource_tfe_workspace.go

```diff
@@ -671,15 +676,54 @@ func resourceTFEWorkspaceDelete(d *schema.ResourceData, meta interface{}) error
 	id := d.Id()
 
 	log.Printf("[DEBUG] Delete workspace %s", id)
-	err := tfeClient.Workspaces.DeleteByID(ctx, id)
+
+	ws, err := tfeClient.Workspaces.ReadByID(ctx, id)
 	if err != nil {
 		if err == tfe.ErrResourceNotFound {
 			return nil
 		}
 		return fmt.Errorf(
-			"Error deleting workspace %s: %w", id, err)
+			"Error reading workspace %s: %w", id, err)
+	}
+
+	forceDelete := d.Get("force_delete").(bool)
+
+	// presence of Permissions.CanForceDelete will determine if current version of TFE supports safe deletes
+	if ws.Permissions.CanForceDelete == nil {
+		if forceDelete {
+			err = tfeClient.Workspaces.DeleteByID(ctx, id)
+		} else {
+			return fmt.Errorf(
+				"Error deleting workspace %s: This workspace must be force deleted by setting force_delete=true", id)
+		}
+	} else if *ws.Permissions.CanForceDelete {
+		if forceDelete {
+			err = tfeClient.Workspaces.DeleteByID(ctx, id)
+		} else {
+			err = tfeClient.Workspaces.SafeDeleteByID(ctx, id)
+			if err != nil {
+				if strings.Contains(err.Error(), "conflict") {
+					return fmt.Errorf(
+						"Error deleting workspace %s: %w\nTo delete this workspace without destroying the managed resources, add force_delete=true to the resource config", id, err)
+				}
+			}
+		}
+	} else {
+		if forceDelete {
+			return fmt.Errorf(
+				"Error deleting workspace %s: missing required permissions to set force_delete=true", id)
```

Comment:
💡 this is a little nitpicky, but the user **does** have permission to set force_delete=true (because adding that to  the config doest delete the workspace)...they just dont have permission to remove a workspace which has `force_delete=true`

I think we could be more explicit here and say something like `Error deleting workspace %s: missing required permissions to set force delete workspaces in the %s organization`